### DISCUSSION TITLE
How to use GPT-4

### DISCUSSION THREAD

**Discussion by julian-hoch** [TheR1D/shell_gpt#342] [General]
I have been using sgpt for a while and it's pretty great. However its using 3.5. On the ChatGPT web interface, I am using GPT4, but I cannot figure out how to do that with sgpt in the shell. It does not seem to accept my attempts to set a model, or at least I am not sure how this expected to work. I tried a variety of shell commands, but none of them work. For example I thought this might work:

```
sgpt --model "gpt-4" "what version are you?"
```

But this only results in HTTPError 404.

**Comment by ivandimitrov8080** [comment]
Same for me as well.

**Comment by David-Else** [comment]
This works great. Are you sure you have access to GPT4, you have to have made a payment to OpenAI before it is enabled.
```sh
alias ai="sgpt"
alias ai4="sgpt --model='gpt-4'"
```

**Comment by catgoose** [comment]
If you make a call to `https://api.openai.com/v1/models` you can see if you have access to gpt-4:

```bash
curl https://api.openai.com/v1/models \
      -H 'Content-Type: application/json' \
      -H 'Authorization: Bearer YOUR_API_KEY'
```

**Comment by ivandimitrov8080** [comment]
I had to reinstall the program for it to work. Or maybe I didn't have access to gpt-4 at the time and it happened to come on as I reinstalled it - can't tell for sure.

**Comment by gitneep** [comment]
I only seem to be able to get GPT-3.5:

```
$ sgpt --model='gpt-4' 'what version are you?'
As an AI developed by OpenAI, I don't have a specific version like software applications do.
However, I'm based on the GPT-3 model, which is the latest version of the Generative Pretrained
Transformer models as of the time of this conversation. Please note that I'm continuously updated
and improved upon by the OpenAI team.

```
I do appear to have access to GPT-4:
```
$ curl https://api.openai.com/v1/models \
>       -H 'Content-Type: application/json' \
>       -H 'Authorization: Bearer [removed]'
{
  "object": "list",
  "data": [
[...]
    {
      "id": "gpt-4-0613",
      "object": "model",
      "created": 1686588896,
      "owned_by": "openai"
    },
[...]
    {
      "id": "gpt-4",
      "object": "model",
      "created": 1687882411,
      "owned_by": "openai"
    },
[...]
```

**Comment by tijszwinkels** [comment]
Use `sgpt --model openai/gpt-4-turbo-preview`
